lib/sycamore/sycamore/connectors/neo4j/neo4j_writer.py

```diff
@@ -0,0 +1,343 @@
+                    "uuid:ID": key,
+                    ":START_ID": value["START_ID"],
+                    ":END_ID": value["END_ID"],
+                    ":TYPE": value["TYPE"],
+                }
+                for key, value in value["properties"].items():
+                    if key not in ["uuid:ID", ":START_ID", ":END_ID", ":TYPE"]:
+                        rel[key] = value
+
+                relationships[value["START_LABEL"]][value["END_LABEL"]].append(rel)
+        return nodes, relationships
+
+    def _write_nodes_csv(self, nodes):
+        for node_label, rels in nodes.items():
+            csv_path = os.path.expanduser(f"{self._import_dir}/sycamore/nodes/{node_label}.csv")
+            headers = None
+            with open(csv_path, mode="r", newline="") as file:
+                csv_reader = csv.reader(file)
+                headers = sorted(set(next(csv_reader)))
+
+            with open(csv_path, "a", newline="") as file:
+                writer = csv.writer(file)
+                for entry in rels:
+                    row = [entry.get(header, "") for header in headers]
+                    writer.writerow(row)
+
+    def _write_relationships_csv(self, relationships):
+        for start_label, end_labels in relationships.items():
+            for end_label, rels in end_labels.items():
+                csv_path = os.path.expanduser(
+                    f"{self._import_dir}/sycamore/relationships/{start_label}_{end_label}.csv"
+                )
+                headers = None
+                with open(csv_path, mode="r", newline="") as file:
+                    csv_reader = csv.reader(file)
+                    headers = sorted(set(next(csv_reader)))
+
+                with open(csv_path, "a", newline="") as file:
+                    writer = csv.writer(file)
+                    for entry in rels:
+                        row = [entry.get(header, "") for header in headers]
+                        writer.writerow(row)
+
+    def write_docs(self, docs: list[Document]) -> list[Document]:
+
+        nodes, relationships = self._parse_docs(docs)
+        self._write_nodes_csv(nodes)
+        self._write_relationships_csv(relationships)
+        return docs
+
+    def _write_docs_tt(self, docs: list[Document]) -> list[Document]:
+        if self._name:
+            with TimeTrace(self._name):
+                return self.write_docs(docs)
+        else:
+            with TimeTrace("UnknownWriter"):
+                return self.write_docs(docs)
+
+
+class Neo4jLoadCSV:
+    def __init__(self, client_params: Neo4jWriterClientParams, target_params: Neo4jWriterTargetParams, **kwargs):
+        self._client = Neo4jWriterClient.from_client_params(client_params)
+        self._nodes = [f for f in os.listdir(client_params.import_dir + "/sycamore/nodes")]
+        self._relationships = [f for f in os.listdir(client_params.import_dir + "/sycamore/relationships")]
+        self._labels = [f[:-4] for f in self._nodes]
```

Comment:
where'd the :-4 come from?

Reply:
oh. dropping `.csv`? Add a comment explaining if so and then don;t you also need to do that for the relationships?

Reply:
yeah dropping the .csv. I do this for labels specifically because for fast performance i implemented neo4j is constraints over node labels. Currently have no need to do this for relationships as I don't really need to index over uuid for relationships.

Reply:
will update with comment

Reply:
updated